### PR TITLE
Added the new improvements in operations and errors

### PR DESCRIPTION
**Pull request by jnicolaslaurenti**
Added the operation with negative numbers, new errors management and concatenation of operation

Pixel 3A:

Calculator_Pixel_pr5

### CODE REVIEW

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 1]
```diff
@@ -22,33 +24,21 @@
...
-        switch (switchOp) {
-            case FIRST_OPERAND: {
-                if (!firstOperand.equals(ZERO_RESULT_STRING)) {
-                    firstOperand += value;
-                } else {
-                    firstOperand = value;
-                }
-                break;
+        if (switchOp == FIRST_OPERAND) {
+            if (firstOperand.isEmpty()) {
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: can you ommit this if logic and just add firstOperand += value?

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 2]
```diff
@@ -22,33 +24,21 @@
...
-            case SECOND_OPERAND: {
-                if (!secondOperand.equals(ZERO_RESULT_STRING)) {
-                    secondOperand += value;
-                } else {
-                    secondOperand = value;
-                }
-                break;
+        } else {
+            switchOp = SECOND_OPERAND;
+            if (!secondOperand.equals(ZERO_RESULT_STRING)) {
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java:38]: same here. is this ok?
can you do this op? `2 - 0.5` ?

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 3]
```diff
@@ -132,22 +138,40 @@ public Double getResult() {
...
+        return aux;
     }
 
     @Override
     public String getLastModified() {
-        if (switchOp == FIRST_OPERAND) {
-            return firstOperand;
-        } else {
-            return secondOperand;
+        String lastModified = "";
```
Comment (BQuinteros) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: pls remove this hardcode

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 4]
```diff
@@ -99,13 +87,30 @@ public void delete() {
             }
     }
 
+    public void manageMinusOperator() {
+        if (firstOperand.isEmpty()) {
```
Comment (BQuinteros) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java:71]: We can use switch here and avoid the concatenation of ifs

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 5]
```diff
@@ -132,22 +138,40 @@ public Double getResult() {
...
         return result;
     }
 
-    @Override
-    public String getOperator() {
-        return operator;
+    public Error getError() {
+        Error aux = error;
+        error = Error.NONE;
+        return aux;
```
Comment (mauroLonghi) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java:145]: split the aux value into the return sentence

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 3 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 4 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java:71]: Can't be solve with a switch

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 1 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 5 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java:145]: Solved!

**Review by mauroLonghi — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by LuchoElissondoGlobant — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**